### PR TITLE
feat: Add stylelint for SCSS linting

### PR DESCRIPTION
**Pull request by yokuze**

### CODE REVIEW

**Comment by yokuze** [comment]
@jthomerson after this is reviewed and merged, we'll need a new version of @silvermine/standardization published.

Ping @kmuncie to give sign-off on the rules. It should be a quick review since the YML file is basically the same as what we had on our internal project.

The only change I made was to disable a rule that was enabled by default in the newer version of stylelint-config-standard: https://stylelint.io/user-guide/rules/no-invalid-position-at-import-rule#no-invalid-position-at-import-rule

I disabled it because there were places that we have `@import` statements interspersed with variable declarations. We do that when we are overriding `!default` variables declared in the files that we are importing. It makes sense to me to keep those variable overrides alongside the imported files instead of putting them all below the `@import` statements as that rule seems to want us to do.

File: package.json [hunk 1]
```diff
@@ -30,7 +30,11 @@
     "@commitlint/travis-cli": "12.1.1",
     "check-node-version": "4.0.3",
     "grunt-markdownlint": "2.9.0",
-    "markdownlint": "0.19.0"
+    "grunt-stylelint": "0.16.0",
+    "markdownlint": "0.19.0",
+    "stylelint": "13.13.1",
+    "stylelint-config-standard": "22.0.0",
```
Comment (kmuncie) [inline package.json:36]: This is 2 major versions newer than what we have been using but I looked at the changes and there is nothing breaking or odd. 👍

File: .stylelintrc.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      - 140
+      - ignorePattern:
+         - /https?://[0-9,a-z]*.*/
+   max-empty-lines: 1
+   string-quotes: single
+   string-no-newline: true
+   color-hex-case: lower
+   color-hex-length: long
+   color-no-invalid-hex: true
+   no-invalid-position-at-import-rule: null
```
Comment (kmuncie) [inline .stylelintrc.yml:29]: This is added from our existing config. Was there a known need or just something noticed that would be helpful?

**Review by kmuncie — COMMENTED** [review]
Thanks @yokuze - looks good, just one question.

File: .stylelintrc.yml (hunk 2 above)
Comment (yokuze) [reply .stylelintrc.yml:29]: See note here: https://github.com/silvermine/standardization/pull/23#issuecomment-839125472

Comment (kmuncie) [reply .stylelintrc.yml:29]: 🤦 lol thanks

**Review by kmuncie — APPROVED** [review]

**Review by jthomerson — APPROVED** [review]

**Pull request merged**